department-of-veterans-affairs/caseflow
# Only display guest link to VSO users

Resolves https://vajira.max.gov/browse/CASEFLOW-1333

### Description
- Make changes to logic in component `StaticVirtualHearing` so it displays different label and link based on who the user is
  - `VLJ Virtual Hearing Link` and host link for users with roles: `Build HearSched` or `Edit HearSched` or if the user is a judge
  - `Virtual Hearing Link` and guest link for all other users
- Make changes to logic in component `HearingLinks` so it omits the VLJ Link if the user isn't a judge or does not have roles: `Build HearSched` or `Edit HearSched`
- small refactors
- jests

### Testing Plan
1. Login an Michaeal_VSO user and go to http://localhost:3000/hearings/schedule
2. Go to docket with virtual hearing
3. Notice that the link for virtual hearing has label `Virtual Hearing Link` and when you click on the link it takes to a guest link

4. Login as a `BVSYELLOW' and go to http://localhost:3000/hearings/schedule
5. Go to docket with virtual hearing
6. Notice that the link for virtual hearing has label `VLJ Virtual Hearing Link` and when you click on the link it takes to a host link
